### PR TITLE
[CI] Use `float16` for `test_bad_words` to ensure compatibility with XPUs

### PR DESCRIPTION
**Pull request by faaany**
Recent updates to the UT model from [PR #2405](https://github.com/vllm-project/vllm/pull/24057) have caused the `test_bad_words` unit test to fail in the XPU CI. The failure is not due to the bad_words feature itself, but rather due to cumulative precision issues introduced by the updated tiny model with BF16 when running on XPUs.

This PR updates the `test_bad_words` unit test to use the float16 data type for the model. The change ensures compatibility and numerical stability on Intel XPUs. Switching to float16 resolves the precision discrepancies observed with float32 on XPUs while preserving the correctness of the test.

The updated test has been validated across all supported platforms (CPU, GPU, and XPU) to ensure consistent behavior.

## Test
```bash
pytest -rA tests/v1/sample/test_sampling_params_e2e.py
```

## Test Result on both CUDA and XPU
```bash
================================================ short test summary info ================================================
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_n_gt_1
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_best_of
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_penalties
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_stop
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_stop_token_ids
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_detokenize_false
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_bad_words
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_logits_processor
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_allowed_token_ids
PASSED tests/v1/sample/test_sampling_params_e2e.py::test_seed
============================================ 10 passed, 2 warnings in 36.30s ============================================
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by jikunshang** [comment]
maybe this due to kernel accuracy issue on bf16  + Intel GPU.
